subliminal/providers/legendastv.py

```diff
@@ -0,0 +1,466 @@
+
+    def __init__(self, username=None, password=None):
+        if username is not None and password is None or username is None and password is not None:
+            raise ConfigurationError('Username and password must be specified')
+
+        self.username = username
+        self.password = password
+        self.logged_in = False
+
+    def initialize(self):
+        self.session = Session()
+
+        # login
+        if self.username is not None and self.password is not None:
+            logger.info('Logging in')
+            data = {'_method': 'POST', 'data[User][username]': self.username, 'data[User][password]': self.password}
+            r = self.session.post('%s/login' % self.server_url, data, allow_redirects=False, timeout=TIMEOUT)
+            r.raise_for_status()
+
+            soup = ParserBeautifulSoup(r.content, ['lxml', 'html.parser'])
+            auth_error = soup.find('div', {'class': 'alert-error'}, text=re.compile(u'.*Usuário ou senha inválidos.*'))
+
+            if auth_error:
+                raise AuthenticationError(self.username)
+
+            logger.debug('Logged in')
+            self.logged_in = True
+
+    def terminate(self):
+        # logout
+        if self.logged_in:
+            logger.info('Logging out')
+            r = self.session.get('%s/users/logout' % self.server_url, timeout=TIMEOUT)
+            r.raise_for_status()
+            logger.debug('Logged out')
+            self.logged_in = False
+
+        self.session.close()
+
+    def matches(self, expected, actual, ignore_episode=False):
+        """
+        Matches two dictionaries (expected and actual). The dictionary keys follow the guessit properties names.
+        If the expected dictionary represents a movie:
+          - ``type`` should match
+          - ``title`` should match
+          - ``year`` should match, unless they're not defined and expected and actual ``title``s are the same
+        If the expected dictionary represents an episode:
+          - ``type`` should match
+          - ``series`` should match
+          - ``eason`` should match
+          - ``episodeNumber`` should match, unless ``ignore_episode`` is True
+
+        :param expected: dictionary that contains the expected values
+        :param actual: dictionary that contains the actual values
+        :param ignore_episode: True if should ignore episodeNumber matching. Default: False
+        :return: True if actual matches expected
+        :rtype: bool
+        """
+        if expected.get('type') != actual.get('type'):
+            return False
+
+        if expected.get('type') == 'movie':
+            if not self.name_matches(expected.get('title'), actual.get('title')):
+                return False
+            if expected.get('year') != actual.get('year'):
+                if expected.get('year') and actual.get('year'):
+                    return False
+                if expected.get('title', '').lower() != actual.get('title', '').lower():
```

Comment:
@fernandog 
All these title manipulations shouldn't be done in subliminal, they should come from guessit.